packages/common/src/converter/payload-converter.ts

```diff
@@ -55,6 +55,20 @@ export function mapToPayloads<K extends string>(converter: PayloadConverter, map
   ) as Record<K, Payload>;
 }
 
+/**
+ * Run {@link PayloadConverter.toPayload} on each value in the map.
+ *
+ * @throws {@link ValueError} if conversion of any value in the map fails
+ */
+export function typedMapToPayloads<K extends string, T>(
+  converter: PayloadConverter,
```

Comment:
AFAICS, you should be able to just add a `T = any` generic parameter, like so:

```
export function mapToPayloads<K extends string, T = any>(
  converter: PayloadConverter,
  map: Record<K, T>
): Record<K, Payload> {
  return Object.fromEntries(
    Object.entries(map).map(([k, v]): [K, Payload] => [k as K, converter.toPayload(v)])
  ) as Record<K, Payload>;
}
```

Then you no longer need the `typedMapToPayloads()` function.